**Task 1.1**

Budi sedang melakukan riset untuk melihat pengaruh suhu harian terhadap minat warga lokal untuk mengunjungi lokasi wisata di musim liburan. Lokasi risetnya adalah di negara bernama Puerto Rico. Budi mengambil data suhu di seluruh region di Puerto Rico selama satu bulan yang termasuk ke dalam musim libur anak sekolah. Dia pergi bersama satu orang asistennya, mahasiswa tingkat awal di universitas tempat dia sedang menempuh studi.

Di sisi lain, Budi juga diminta oleh professor di tempat dia kuliah untuk mengumpulkan data temperatur secara detil setiap 15 menit di ibu kota Puerto Rico. Data ini nantinya akan digunakan oleh professor untuk melakukan suatu eksperimen di bidang Engineering.

Saat kembali ke Indonesia, Budi sadar bahwa dia melakukan pengukuran suhu dengan menggunakan satuan Celcius untuk setiap data. Di sisi lain, satuan yang biasa dipakai di Puerto Rico adalah Fahrenheit. Professor juga memerlukan data tersebut dalam satuan Kelvin untuk penelitiannya di bidang Engineering. Saat hendak mengkonversi data tersebut, Budi sadar bahwa asistennya salah menginput tipe data untuk beberapa data. Ada sebagian data yang berada dalam format teks (string), yang mana seharusnya formatnya adalah bilangan desimal. 

1. Buat satu function untuk mengkonversi suhu dari satuan Celcius menjadi Fahrenheit. Beri nama functionnya sebagai **celcius_to_fahrenheit**
2. Buat satu function untuk mengkonversi suhu dari satuan Celcius menjadi Kelvin. Beri nama functionnya sebagai **celcius_to_kelvin**
3. Pastikan bahwa function tetap dapat mengkonversi suhu dengan benar, walaupun input datanya adalah teks.
4. Buat hasil akhirnya sesuai dengan kunci jawaban yang tersedia di bawah, berada dalam foramt **Temperatur : 86.0 Fahrenheit**

In [11]:
# celcius to fahrenheit

def celcius_to_fahrenheit():
    try:
        celcius = input("Input temp °C: ")
        _celc = float(celcius)
        c2f = (9/5*_celc)+32
        print(f"Nilai suhu saat ini {c2f} °F")
    except ValueError:
        print("Input not valid, please input number")

def celcius_to_kelvin():
    try:
        celcius = input("input temp °C: ")
        _celc = float(celcius)
        c2k = _celc+273.15
        print(f"Nilai suhu saat ini {c2k} °K")
    except ValueError:
        print("Input not valid, please input number")



Nilai suhu saat ini 293.15 °F


In [2]:
celcius_to_fahrenheit()

NameError: name 'celcius_to_fahrenheit' is not defined

Temperatur : 86.0 Fahrenheit

In [4]:
# check your answer

celcius_to_fahrenheit() #93.1

Nilai suhu saat ini 199.57999999999998 °F


Temperatur : 199.57999999999998 Fahrenheit

In [6]:
# check your answer

celcius_to_fahrenheit() #-27.5

Nilai suhu saat ini -17.5 °F


Temperatur : -17.5 Fahrenheit

In [14]:
celcius_to_kelvin() #100

Nilai suhu saat ini 373.15 °F


Temperatur : 373.15 Kelvin

In [1]:
celcius_to_kelvin() #-59

NameError: name 'celcius_to_kelvin' is not defined

Temperatur : 178.15 Kelvin

In [16]:
celcius_to_kelvin()

Nilai suhu saat ini 287.15 °F


Temperatur : 287.15 Kelvin

**Task 1.2**

Seorang Data Analyst sedang mengobservasi variabel apa saja yang bisa mempengaruhi harga suatu laptop. Dia juga mau mengetahui perbedaan harga laptop di brand yang berbeda dengan spesifikasi yang berbeda pula.

1. Import data dalam format CSV
2. Pastikan data pada semua kolom berada dalam format yang sesuai. Jika belum, maka ubah formatnya (data type)
3. Cari tahu Median, Mean, Maximum Price, dan Minimum Price dari seluruh harga laptop
4. Cari tahu Mean dan Maximum Price untuk masing-masing brand
5. Apakah berat laptop mempengaruhi harga suatu laptop?
6. Brand apa yang menjual laptop dengan harga yang paling ekonomis menurut analisis Anda?